compiler/ras/ILValidator.cpp

```diff
@@ -21,333 +21,132 @@
+                 delete_pointed_object<TR::NodeValidationRule>);
    }
 
-void TR::ILValidator::soundnessRule(TR::TreeTop *location, bool condition, const char *formatStr, ...)
-   {
-   if (!condition)
-      {
-      if (location && location->getNode())
-         printDiagnostic("*** VALIDATION ERROR: IL is unsound at n%dn ***\nMethod: %s\n", location->getNode()->getGlobalIndex(), comp()->signature());
-      else
-         printDiagnostic("*** VALIDATION ERROR: IL is unsound ***\nMethod: %s\n", comp()->signature());
-      va_list args;
-      va_start(args, formatStr);
-      vprintDiagnostic(formatStr, args);
-      va_end(args);
-      printDiagnostic("\n");
-      FAIL();
-      }
-   }
 
-void TR::ILValidator::validityRule(Location &location, bool condition, const char *formatStr, ...)
-   {
-   if (!condition)
-      {
-      _isValidSoFar = false;
-      TR::Node *node = location.currentNode();
-      printDiagnostic("*** VALIDATION ERROR ***\nNode: %s n%dn\nMethod: %s\n", node->getOpCode().getName(), node->getGlobalIndex(), comp()->signature());
-      va_list args;
-      va_start(args, formatStr);
-      vprintDiagnostic(formatStr, args);
-      va_end(args);
-      printDiagnostic("\n");
-      FAIL();
-      }
-   }
 
-void TR::ILValidator::updateNodeState(Location &newLocation)
+TR::Compilation *TR::ILValidator::comp()
    {
-   TR::Node  *node = newLocation.currentNode();
-   NodeState &state = _nodeStates[node];
-   if (node->getReferenceCount() == state._futureReferenceCount)
-      {
-      // First occurrence -- do some bookkeeping
-      //
-      if (node->getReferenceCount() == 0)
-         {
-         validityRule(newLocation, node->getOpCode().isTreeTop(), "Only nodes with isTreeTop opcodes can have refcount == 0");
-         }
-      else
-         {
-         _liveNodes.add(node);
-         }
-      }
-
-   if (_liveNodes.contains(node))
-      {
-      validityRule(newLocation, state._futureReferenceCount >= 1, "Node already has reference count 0");
-      if (--state._futureReferenceCount == 0)
-         {
-         _liveNodes.remove(node);
-         }
-      }
-   else
-      {
-      validityRule(newLocation, node->getOpCode().isTreeTop(), "Node has already gone dead");
-      }
-
-   if (isLoggingEnabled())
-      {
-      static const char *traceLiveNodesDuringValidation = feGetEnv("TR_traceLiveNodesDuringValidation");
-      if (traceLiveNodesDuringValidation && !_liveNodes.isEmpty())
-         {
-         traceMsg(comp(), "    -- Live nodes: {");
-         char *separator = "";
-         for (LiveNodeWindow::Iterator lnwi(_liveNodes); lnwi.currentNode(); ++lnwi)
-            {
-            traceMsg(comp(), "%sn%dn", separator, lnwi.currentNode()->getGlobalIndex());
-            separator = ", ";
-            }
-         traceMsg(comp(), "}\n");
-         }
-      }
-
+   return _comp;
    }
 
-bool TR::ILValidator::treesAreValid(TR::TreeTop *start, TR::TreeTop *stop)
+bool TR::ILValidator::validate()
```

Comment:
Given that the return value here is never checked, and the complexity that is injected below trying to figure out what to return, perhaps the return type should just be removed, the returns removed.